chrome/content/zotero/xpcom/data/item.js

```diff
@@ -2889,20 +2891,24 @@ Zotero.defineProperty(Zotero.Item.prototype, 'attachmentPath', {
 		}
 		
 		if (linkMode == Zotero.Attachments.LINK_MODE_LINKED_FILE) {
-			if (this._libraryID) {
-				let libraryType = Zotero.Libraries.get(this._libraryID).libraryType;
+			let libraryID = this._libraryID;
+			if (libraryID) {
+				let libraryType = Zotero.Libraries.get(libraryID).libraryType;
 				if (libraryType != 'user') {
 					throw new Error("Linked files can only be added to user library");
 				}
+			} else {
+				Zotero.debug(`'this._libraryID' has not been set, defaulting to 'userLibraryID'`);
+				libraryID = Zotero.Libraries.userLibraryID;
```

Comment:
This is one of the biggest changes.  This was done to appease a lot of the unit tests that don't otherwise specify any `libraryID` or provided a fully-fleshed out `parentItem` that also doesn't have a `libraryID` setup.